src/pudl/analysis/mcoe.py

```diff
@@ -474,40 +467,37 @@ def mcoe(
         .pipe(pudl.validate.no_null_cols, df_name="fuel_cost + capacity_factor")
     )
 
-    # Combine MCOE derived values with all the generator attributes:
-    mcoe_out = (
-        pd.merge(
-            left=(
-                pudl_out.gens_eia860()
-                .assign(year=lambda x: x.report_date.dt.year)
-                .drop("report_date", axis="columns")
-            ),
-            right=mcoe_out.assign(year=lambda x: x.report_date.dt.year),
-            # This "how" determines whether MCOE or gens_eia860 is the backbone
-            how="left" if all_gens else "right",
-            on=["year", "plant_id_eia", "generator_id"],
-        )
-        .astype({"year": str})
-        .assign(report_date=lambda x: x.report_date.fillna(pd.to_datetime(x.year)))
-        .drop("year", axis="columns")
-        .pipe(pudl.validate.no_null_rows, df_name="mcoe_all_gens", thresh=0.9)
-    )
+    # Combine MCOE derived values with generator attributes
+    default_gens_cols = [
+        "plant_id_eia",
+        "generator_id",
+        "report_date",
+        "unit_id_pudl",
+        "plant_id_pudl",
+        "plant_name_eia",
+        "utility_id_eia",
+        "utility_id_pudl",
+        "utility_name_eia",
+    ]
+    if all_gens_cols:
+        gens = pudl_out.gens_eia860()
+    else:
+        gens = pudl_out.gens_eia860()[default_gens_cols + extra_gens_cols]
```

Comment:
If `extra_gens_cols` can be either `all` or a list of strings (as I suggest below to avoid ambiguity) then the flow control would be:

```py
if extra_gens_cols == "all":
    gens = pudl_out.gens_eia860()
else:
    gens = pudl_out.gens_eia860()[default_gens_cols + extra_gens_cols]
```